pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -975,10 +951,13 @@ func (nc *Controller) tryUpdateNodeHealth(node *v1.Node) (time.Duration, v1.Node
 				}
 			}
 		}
+		if currentReadyCondition != nil {
```

Comment:
Why only if nil?
If it wasn't nil we could have also updated it to ConditionUnknown.

Reply:
Previous logic only updated currentReadyCondition when it wasn't nil. Here I wanted to have the same results of the function. If we update the currentReadyCondition to something different then nil, it will trigger different if branch - https://github.com/kubernetes/kubernetes/blob/944f9ce3a18000dc2be3f331e2787a8b328367a8/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L721

Reply:
OK - I see. But that's a microoptimization.
Let's do that every time - it will be easier to reason about.

Reply:
I'm not sure if is an optimization or assumption to not make decision based on fake condition https://github.com/kubernetes/kubernetes/blob/944f9ce3a18000dc2be3f331e2787a8b328367a8/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L814-L816

Assuming that we set currentReadyCondition to something different than nil, it will trigger pod eviction due to observedReadyCondition.Status == v1.ConditionUnknown https://github.com/kubernetes/kubernetes/blob/944f9ce3a18000dc2be3f331e2787a8b328367a8/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L750

The question is if it should?

Reply:
OK - it's returned later.

> The question is if it should?

This means, that Kubelet never reported its status and we reached this path because corresponding Lease was sent. in that case, nothing really should be running on that node, so I don't think it really changes much...

Reply:
Done. `if` removed.